# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasource/gtest/TestContainerParser.cpp
# Commit: f6df4eb84932
# Full Hash: f6df4eb84932ddaccae4e53d4897234e0153913e
# Author: Chris Pearce <cpearce@mozilla.com>
# Date: 2019-06-14 10:00:22
# Regressor Bug: 1558364
# File Overlap Count: 1
# Description:
#   Bug 1558364 - Add MediaSpan and use it for TrackBuffersManager::mInputBuffer. r=jya
#   
#   As seen in this profile of a Twitch replay: https://perfht.ml/2K9Ydb3 we can
#   often end up spending time in TrackBuffersManager::CodedFrameProcessing()
#   shaving off bytes from the front off TrackBuffersManager::mInputBuffer. This
# ==============================================================================

diff -r e708fe457a95 -r f6df4eb84932 dom/media/mediasource/gtest/TestContainerParser.cpp
--- a/dom/media/mediasource/gtest/TestContainerParser.cpp	Fri Jun 14 00:37:25 2019 +0000
+++ b/dom/media/mediasource/gtest/TestContainerParser.cpp	Fri Jun 14 02:10:09 2019 +0000
@@ -45,46 +45,46 @@
 
   // Test a valid header.
   RefPtr<MediaByteBuffer> header = make_adts_header();
-  EXPECT_TRUE(NS_SUCCEEDED(parser->IsInitSegmentPresent(header)));
+  EXPECT_TRUE(NS_SUCCEEDED(parser->IsInitSegmentPresent(MediaSpan(header))));
 
   // Test variations.
   uint8_t save = header->ElementAt(1);
   for (uint8_t i = 1; i < 3; ++i) {
     // Set non-zero layer.
     header->ReplaceElementAt(1, (header->ElementAt(1) & 0xf9) | (i << 1));
-    EXPECT_FALSE(NS_SUCCEEDED(parser->IsInitSegmentPresent(header)))
+    EXPECT_FALSE(NS_SUCCEEDED(parser->IsInitSegmentPresent(MediaSpan(header))))
         << "Accepted non-zero layer in header.";
   }
   header->ReplaceElementAt(1, save);
   save = header->ElementAt(2);
   header->ReplaceElementAt(2, (header->ElementAt(2) & 0x3b) | (15 << 2));
-  EXPECT_FALSE(NS_SUCCEEDED(parser->IsInitSegmentPresent(header)))
+  EXPECT_FALSE(NS_SUCCEEDED(parser->IsInitSegmentPresent(MediaSpan(header))))
       << "Accepted explicit frequency in header.";
   header->ReplaceElementAt(2, save);
 
   // Test a short header.
   header->SetLength(6);
-  EXPECT_FALSE(NS_SUCCEEDED(parser->IsInitSegmentPresent(header)))
+  EXPECT_FALSE(NS_SUCCEEDED(parser->IsInitSegmentPresent(MediaSpan(header))))
       << "Accepted too-short header.";
-  EXPECT_FALSE(NS_SUCCEEDED(parser->IsMediaSegmentPresent(header)))
+  EXPECT_FALSE(NS_SUCCEEDED(parser->IsMediaSegmentPresent(MediaSpan(header))))
       << "Found media segment when there was just a partial header.";
 
   // Test a header with short data.
   header = make_adts_header();
   header->AppendElements(1);
-  EXPECT_TRUE(NS_SUCCEEDED(parser->IsInitSegmentPresent(header)))
+  EXPECT_TRUE(NS_SUCCEEDED(parser->IsInitSegmentPresent(MediaSpan(header))))
       << "Rejected a valid header.";
-  EXPECT_TRUE(NS_SUCCEEDED(parser->IsMediaSegmentPresent(header)))
+  EXPECT_TRUE(NS_SUCCEEDED(parser->IsMediaSegmentPresent(MediaSpan(header))))
       << "Rejected a one-byte media segment.";
 
   // Test parse results.
   header = make_adts_header();
-  EXPECT_FALSE(NS_SUCCEEDED(parser->IsMediaSegmentPresent(header)))
+  EXPECT_FALSE(NS_SUCCEEDED(parser->IsMediaSegmentPresent(MediaSpan(header))))
       << "Found media segment when there was just a header.";
   int64_t start = 0;
   int64_t end = 0;
-  EXPECT_TRUE(
-      NS_FAILED(parser->ParseStartAndEndTimestamps(header, start, end)));
+  EXPECT_TRUE(NS_FAILED(
+      parser->ParseStartAndEndTimestamps(MediaSpan(header), start, end)));
 
   EXPECT_TRUE(parser->HasInitData());
   EXPECT_TRUE(parser->HasCompleteInitData());
@@ -111,22 +111,22 @@
 
   // Test the header only.
   RefPtr<MediaByteBuffer> header = make_adts_header();
-  EXPECT_TRUE(NS_SUCCEEDED(parser->IsInitSegmentPresent(header)));
+  EXPECT_TRUE(NS_SUCCEEDED(parser->IsInitSegmentPresent(MediaSpan(header))));
 
   // Test with the correct length of (invalid) frame data.
   size_t header_length = header->Length();
   size_t data_length = 24;
   size_t frame_length = header_length + data_length;
   header->AppendElements(data_length);
-  EXPECT_TRUE(NS_SUCCEEDED(parser->IsInitSegmentPresent(header)))
+  EXPECT_TRUE(NS_SUCCEEDED(parser->IsInitSegmentPresent(MediaSpan(header))))
       << "Rejected a valid header.";
-  EXPECT_TRUE(NS_SUCCEEDED(parser->IsMediaSegmentPresent(header)))
+  EXPECT_TRUE(NS_SUCCEEDED(parser->IsMediaSegmentPresent(MediaSpan(header))))
       << "Rejected a full (but zeroed) media segment.";
   int64_t start = 0;
   int64_t end = 0;
   // We don't report timestamps from ADTS.
-  EXPECT_TRUE(
-      NS_FAILED(parser->ParseStartAndEndTimestamps(header, start, end)));
+  EXPECT_TRUE(NS_FAILED(
+      parser->ParseStartAndEndTimestamps(MediaSpan(header), start, end)));
   EXPECT_EQ(start, 0);
   EXPECT_EQ(end, 0);
 